demos/KINETIS/RT-FREEDOM-KL25Z-SHELL/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+    Copyright (C) 2016 Jonathan Struebel
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+#include "hal.h"
+
+#include "shellcfg.h"
+
+SerialConfig s0cfg = {
+  115200
+};
+
+/*
+ * Blue LED blinker thread, times are in milliseconds.
+ */
+static THD_WORKING_AREA(waBlinkThread, 128);
+static THD_FUNCTION(BlinkThread, arg) {
+
+  (void)arg;
+
+  chRegSetThreadName("blinker");
+  while (true) {
+    systime_t time = 500;

```

Comment:
Although this will be optimized by the compiler, you would usually set the variable outside the loop!